src/sync.rs

```diff
@@ -1290,6 +1600,56 @@ mod tests {
         });
     }
 
+    #[cfg(all(Py_3_14, not(Py_LIMITED_API)))]
+    #[cfg(not(target_arch = "wasm32"))] // We are building wasm Python with pthreads disabled
+    #[test]
+    fn test_critical_section_mutex2_two_containers() {
+        let (m1, m2) = (PyMutex::new(vec![1, 2, 3]), PyMutex::new(vec![4, 5]));
+
```

Comment:
I wonder, would the test be more interesting if we lock m1 and / or m2 here, and then below the second `s.spawn` call we add something like

```rust
// the other threads waiting for locks should not block this attach
Python::attach(|py| {
    // on free-threaded build, the critical sections should have blocked
    // the other threads from modification.
    #[cfg(Py_GIL_DISABLED)] 
    {
        assert_eq!(&*m1_guard, &[1, 2, 3]);
        assert_eq!(&*m2_guard, &[4, 5]);
    }

    drop(m1_guard);
    drop(m2_guard);
});
```

... so that way, the two threads both will wait until the locks are released? We should still be deadlock free, I believe.


```suggestion
let m1_guard = m1.lock().unwrap();
let m2_guard = m2.lock().unwrap();
```